xbmc/filesystem/CurlFile.cpp

```diff
@@ -902,13 +904,13 @@ bool CCurlFile::Download(const std::string& strURL, const std::string& strFileNa
 // Detect whether we are "online" or not! Very simple and dirty!
 bool CCurlFile::IsInternet()
 {
-  CURL url("http://www.msftncsi.com/ncsi.txt");
+  CURL url(StringUtils::Format("http://%s.msftncsi.com/ncsi.txt", g_application.getNetwork().GetFirstConnectedFamily() == AF_INET6 ? "ipv6" : "www"));
```

Comment:
@Memphiz 

missed that question previously. sorry.

fortunately this won't happen. It is commented here https://github.com/mk01/xbmc/commit/b2564e1bd6b83830eb84771f403935a212d9202b#diff-1d114205997f1293481d13a7f07fa111R102 .

0. IPv4 only host - nothing changed.
1. dual-stack - that means there is IPv4 configuration - and we will get IPv4 configured values first. like case 0.
2. only scenario we get IPv6 first is when host is IPv6 configure only. whether correctly or not is user issue.

(the test will be ok also in 6to4 configurations (IPv6 tunnelled via IPv4 gateway). and in case of 4over6, IPv4 configuration is TUN while IPv6 native ethernet, that's why IPv6 will be chosen for test - but not tested this variant).